tests/test_preprints.py

```diff
@@ -727,3 +728,27 @@ def test_no_call_async_update_on_400_failure(self, requests, mock_async, mock_ma
         update_preprint_share(self.preprint)
         assert not mock_async.called
         assert mock_mail.called
+
+class TestPreprintConfirmationEmails(OsfTestCase):
+    def setUp(self):
+        super(TestPreprintConfirmationEmails, self).setUp()
+        self.user = AuthUserFactory()
+        self.write_contrib = AuthUserFactory()
+        self.project = ProjectFactory(creator=self.user)
+        self.project.add_contributor(self.write_contrib, permissions=[permissions.WRITE])
+        self.preprint = PreprintFactory(project=self.project, is_published=False)
+        contributor_added.connect(notify_added_contributor)
+
+    @mock.patch('website.mails.send_mail')
+    def test_creator_gets_email(self, send_mail):
```

Comment:
This and the below test can be combined-- test failed published first and then real publish.